app/src/main/java/com/github/se/studybuddies/ui/settings/CreateAccount.kt

```diff
@@ -17,59 +17,54 @@ import androidx.compose.runtime.mutableStateOf
-    val uid = userViewModel.getCurrentUserUID()
+  val uid = userViewModel.getCurrentUserUID()
 
-    val email = FirebaseAuth.getInstance().currentUser?.email ?: ""
-    val usernameState = remember { mutableStateOf("") }
-    val photoState = remember { mutableStateOf(Uri.EMPTY) }
+  val email = FirebaseAuth.getInstance().currentUser?.email ?: ""
+  val usernameState = remember { mutableStateOf("") }
+  val photoState = remember { mutableStateOf(Uri.EMPTY) }
 
-    LaunchedEffect(key1 = true) {
-        val defaultProfilePictureUri = withContext(Dispatchers.IO) { userViewModel.getDefaultProfilePicture() }
-        photoState.value = defaultProfilePictureUri
-    }
+  LaunchedEffect(key1 = true) {
+    val defaultProfilePictureUri =
+        withContext(Dispatchers.IO) { userViewModel.getDefaultProfilePicture() }
+    photoState.value = defaultProfilePictureUri
+  }
 
-    val getContent = rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
-        uri?.let { profilePictureUri ->
-            photoState.value = profilePictureUri
-        }
-    }
+  val getContent =
+      rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
+        uri?.let { profilePictureUri -> photoState.value = profilePictureUri }
+      }
 
-    Column(modifier = Modifier
-        .fillMaxSize()
-        .testTag("createScreen")) {
-        LazyColumn(
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(horizontal = 20.dp, vertical = 20.dp),
-            verticalArrangement = Arrangement.spacedBy(20.dp, Alignment.Top),
-            horizontalAlignment = Alignment.CenterHorizontally) {
-            item {
-                Column(
-                    modifier = Modifier.fillMaxWidth(),
-                    verticalArrangement = Arrangement.spacedBy(20.dp)
-                ) {
-                    Text("You signed in with the email address $email")
-                    Spacer(modifier = Modifier.padding(20.dp))
-                    AccountFields(usernameState)
-                    Spacer(modifier = Modifier.padding(20.dp))
-                    SetProfilePicture(photoState) { getContent.launch("image/*") }
-                    SaveButton(usernameState) {
-                        userViewModel.createUser(uid, email, usernameState.value, photoState.value)
-                        navigationActions.navigateTo(Route.GROUPSHOME)
-                    }
+  Column(modifier = Modifier.fillMaxSize()) {
+    LazyColumn(
+        modifier = Modifier.fillMaxWidth().padding(horizontal = 20.dp, vertical = 20.dp),
+        verticalArrangement = Arrangement.spacedBy(20.dp, Alignment.Top),
+        horizontalAlignment = Alignment.CenterHorizontally) {
+          item {
+            Column(
+                modifier = Modifier.fillMaxWidth(),
+                verticalArrangement = Arrangement.spacedBy(20.dp)) {
+                  Text("You signed in with the email address $email")
```

Comment:
It is a good practice to extract all hard coded string that are displayed to the user inside the strings.xml resource file. That way if some day you decide to localize your app and provide a French version for example, you only need to create a french version of the strings.xml file without having to navigate the whole code base to extract each string to this file.

Here is the link to the [localization page](https://developer.android.com/guide/topics/resources/localization#resource-precedence) in the android documentation.

Reply:
That is great, I didn't know about this! Thanks for the tip.